lib/iris/experimental/stratify.py

```diff
@@ -0,0 +1,178 @@
+# (C) British Crown Copyright 2017, Met Office
+#
+# This file is part of Iris.
+#
+# Iris is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Iris is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Iris.  If not, see <http://www.gnu.org/licenses/>.
+"""
+Routines for putting data on new strata (aka. isosurfaces), often in the
+Z direction.
+
+"""
+import copy
+from functools import partial
+
+import numpy as np
+import stratify
+
+from iris.cube import Cube
+from iris.coords import Coord, AuxCoord, DimCoord
+
+
+def _copy_coords_without_dim(cube, result, z_dim):
+    """
+    Helper function to copy across non-z coordinates between cubes.
+
+    Parameters
+    ----------
+    cube : iris.cube.Cube instance
+        Incoming cube containing the coordinates to be copied.
+    result : iris.cube.Cube instance
+        Incoming cube for the coordinates to be copied to.
+    z_dim : Dimension within the cube of the z-dimension
+        The dimension will not be copied, as the incoming z-dimension
+        is model level_height, whilst the outgoing z-dimension is pressure.
+
+    Returns
+    -------
+    Cube
+        A single cube with the non-z dimensions added to
+        incoming cube.
+
+    """
+    # Copy across non-Z coordinates.
+    for coord in cube.dim_coords:
+        dim, = cube.coord_dims(coord)
+        if dim != z_dim:
+            result.add_dim_coord(coord.copy(), dim)
+    for coord in cube.aux_coords:
+        dims = cube.coord_dims(coord)
+        if z_dim not in dims:
+            result.add_aux_coord(coord.copy(), dims)
+    for coord in cube.derived_coords:
+        dims = cube.coord_dims(coord)
+        if z_dim not in dims:
+            result.add_aux_coord(coord.copy(), dims)
+
+
+def relevel(phenom, level_data, target_levels, interp_direction=0,
```

Comment:
@pelson I think that kwarg name `interp_direction` is a little unclear to users as to its intent. 

It's really the `axis` of interpolation relative to the `phenom`, isn't it? The difference from the use of `axis` in `stratify` is that here the user has the convenience of passing a `phenom` coordinate .... indeed they should be able to also the name of a coordinate.